server/src/test/java/org/elasticsearch/discovery/DiscoveryDisruptionIT.java

```diff
@@ -98,7 +99,8 @@ public void testIsolatedUnicastNodes() throws Exception {
      * The rejoining node should take this master node and connect.
      */
     public void testUnicastSinglePingResponseContainsMaster() throws Exception {
-        List<String> nodes = startCluster(4, -1, new int[]{0});
+        List<String> nodes = startCluster(4, -1);
+        internalCluster().setHostsListContainsOnlyFirstNode(true);
```

Comment:
I think this has to happen before you start the cluster, or else the cluster will start with full knowledge.

Reply:
Right fixing